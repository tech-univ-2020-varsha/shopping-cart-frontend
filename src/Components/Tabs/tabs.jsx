import React, { useState } from 'react';
import propTypes from 'prop-types';
import * as styles from './tabs.module.css';
import Categories from '../CategoriesData.jsx/categories';

const clickEvent = (products, setData, setFilterSelected, filter) => {
  setFilterSelected(filter);
  setData(products);
};
const Tabs = ({ products, filterProduct }) => {
  const filterNames = Object.keys(filterProduct);
  const [data, setData] = useState(products);
  //   setData(products);
  const [filterSelected, setFilterSelected] = useState('All');

  //   setData(products);
  return (

    <div className={styles.Container}>
      <div className={styles.filterBar}>
        Filter Categories
        <div className={styles.filterTabs}>
          <div
            key="All"
            className={styles.tabItem}
            onClick={() => clickEvent(products, setData, setFilterSelected, 'All')}
          >
            All
          </div>
          {
         filterNames.map((filter) => (
           <div
             key={filter}
             className={styles.tabItem}
             onClick={() => clickEvent(filterProduct[filter], setData, setFilterSelected, filter)}
           >
             {filter}
           </div>
         ))
          }
        </div>

      </div>

      <Categories data={data} filter={filterSelected} />
    </div>
  );
};

Tabs.propTypes = {
  products: propTypes.arrayOf(propTypes.object).isRequired,
  filterProduct: propTypes.objectOf(propTypes.array).isRequired,
};

export default Tabs;
