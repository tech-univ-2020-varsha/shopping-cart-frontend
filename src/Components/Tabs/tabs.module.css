.filterBar{
    display: flex;
    margin: 3em;
    text-align: center;
    padding: 2em;
}

.filterTabs{
    display: flex;
    justify-content: space-evenly;
    padding-left: 1em;
    border-top-right-radius: 50px 20px;
    margin-left: 2em;
    padding-right: 1em;
    align-items: center;
    border-radius: 10px;
    background-color: rgb(196,196,196);
    padding-left: 2em;
    box-shadow: 5px 5px 5px rgba(68,68,68,0.6);
}
.tabItem{
    display: flex;
    padding: 1em;
    cursor: pointer;
    /* border-radius: 10px; */
    /* margin: 2em; */
    background-color: rgb(196,196,196);
   
}

.Container{
    display: flex;
    flex-direction: column;
}