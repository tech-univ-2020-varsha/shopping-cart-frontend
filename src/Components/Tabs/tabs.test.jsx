import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Tabs from './tabs';

describe('The Tabs component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Tabs products={[]} filterProduct={{}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
