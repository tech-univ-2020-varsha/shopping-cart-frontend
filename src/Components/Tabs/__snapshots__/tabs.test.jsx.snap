// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      Filter Categories
      <div>
        <div>
          All
        </div>
      </div>
    </div>
    <div>
      <div>
        All
      </div>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
